server/src/main/java/org/elasticsearch/indices/NodeIndicesStats.java

```diff
@@ -172,23 +187,33 @@ public ShardCountStats getShardCount() {
         return stats.getShards();
     }
 
+    @Nullable
+    public NodeMappingStats getNodeMappingStats() {
+        return stats.getNodeMappings();
+    }
+
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         stats.writeTo(out);
         out.writeMap(statsByShard, (o, k) -> k.writeTo(o), StreamOutput::writeList);
+        if (out.getVersion().onOrAfter(VERSION_SUPPORTING_STATS_BY_INDEX)) {
+            out.writeMap(statsByIndex, (o, k) -> k.writeTo(o), (o, v) -> v.writeTo(o));
+        }
     }
 
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         NodeIndicesStats that = (NodeIndicesStats) o;
-        return Objects.equals(stats, that.stats) && Objects.equals(statsByShard, that.statsByShard);
+        return Objects.equals(stats, that.stats)
```

Comment:
NIT: Actually non of these are nullable, we could just do explicit `stats.equals(` etc. here.

Reply:
True, although Object.equals does that already. I guess you are saying it to make it faster? Will change this in next commits

Reply:
The faster is irrelevant here since this is only used in tests anyway :) It was more of a cleanup thing.